### ISSUE TITLE
Track deferred #1278 coverage-bound hardening nits

### ISSUE THREAD

**Issue by rjmurillo** [rjmurillo/moq.analyzers#1322]
Related to #1278 and PR #1319.

Two #1278 hardening nits were deferred because they cannot meet the current 100% shipping-diff-coverage bar without a separate design decision.

## Deferred item (e): KnownSymbols Lazy type caching

The requested KnownSymbols type-property caching changed `EventHandler1 => _eventHandler1.Value`. Release coverage could not cover that line through shipping assemblies.

Local analysis found:
- `KnownSymbols.EventHandler1` is only read by `IsEventHandlerDelegate` in `src/Common/NamedTypeSymbolExtensions.cs`.
- `IsEventHandlerDelegate` has no production callers. `rg -n "IsEventHandlerDelegate" src/` returns only its definition.
- Its former job appears covered by `IsEventHandlerShaped`, which reads `EventHandler` and `EventArgs`, not `EventHandler1`.

Decision needed: either remove the dead `IsEventHandlerDelegate`/`EventHandler1` path, or keep it and add a real production caller before caching it.

## Deferred item (b): event-name [NotNullWhen] refactor

The `[NotNullWhen(true)]` refactor for `TryGetEventNameFromLambdaSelector` introduced a defensive false branch:

```csharp
eventName = null;
return false;
```

A verifier probe tried to force Moq1202 through `GetEventNameFromSelector` with a selector that did not resolve to an event symbol. Result: no diagnostic was produced. This matches the production flow: Moq1202/Moq1204 must first resolve the event type to compare event arguments. The event-name lookup uses the same event-symbol resolver, so a diagnostic path that resolves the event type but fails to resolve the event name appears unreachable.

Decision needed: either keep the current suppression, or refactor the API in a way that does not add an unreachable shipping line under the current coverage gate.
